composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,325 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log training metadata to [neptune.ai](https://neptune.ai/)."""
+
+__all__ = ['NeptuneLogger']
+
+import os
+import pathlib
+import warnings
+from functools import partial
+from typing import TYPE_CHECKING, Any, Dict, Optional, Sequence, Set, Union
+
+import numpy as np
+import torch
+
+from composer._version import __version__
+from composer.loggers import LoggerDestination
+from composer.utils import MissingConditionalImportError, dist
+
+if TYPE_CHECKING:
+    from composer import Logger
+    from composer.core import State
+
+
+class NeptuneLogger(LoggerDestination):
+    """Log to `neptune.ai <https://neptune.ai/>`_.
+
+    Args:
+        project (str, optional): The name of your Neptune project,
+            in the form "workspace-name/project-name". If you leave it empty, the
+            ``NEPTUNE_PROJECT`` environment variable will be used.
+        api_token (str, optional): Your Neptune API token.
+            You can leave out this argument if you save your token to the
+            ``NEPTUNE_API_TOKEN`` environment variable (recommended).
+            You can find your API token in the user menu of the Neptune web app.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            Default: ``True``.
+        upload_artifacts (bool, optional): Whether the logger should upload artifacts to Neptune.
+            Default: ``False``.
+        base_namespace (str, optional): The name of the base namespace to log the metadata to.
+            Default: "training".
+        neptune_kwargs (Dict[str, Any], optional): Any additional keyword arguments to the
+            ``neptune.init_run()`` function. For options, see the
+            `Run API reference <https://docs.neptune.ai/api/neptune/#init_run>`_ in the
+            Neptune docs.
+
+    Example:
+        Default Neptune logger:
+            from composer import Trainer
+            from composer.loggers import NeptuneLogger
+            neptune_logger = NeptuneLogger()
+            trainer = Trainer(loggers=neptune_logger)
+            trainer.fit()
+
+        More options:
+            from composer import Trainer
+            from composer.loggers import NeptuneLogger
+            neptune_logger = NeptuneLogger(
+                project="ml-team/classification",
+                upload_artifacts=True,
+                dependencies="infer")  # Neptune Run kwarg
+            trainer = Trainer(loggers=neptune_logger)
+            trainer.fit()
+
+    For more, see the [Neptune-Composer integration guide](https://docs.neptune.ai/integrations/composer/).
```

Comment:
@AleksanderWWW -- are doctests still failing? (Not sure how to check this)

I'm noticing we have still Markdown in some places, which I can change to reST if the issues persist. If that doesn't help, we can try removing all of the `Examples` sections from the docstrings.

Reply:
Not sure, I've some troubles running the check locally and the workflow in GH actions requires approval from code owners :(